### PR TITLE
MET-2497: Handle HTML HUD on project import

### PR DESCRIPTION
**Pull request by mwojcik-sdh**
This PR restructures old projects HTML hud for new structure

### CODE REVIEW

File: src/assets/scripts/ProjectManager/Scene/HTMLElement.js [hunk 1]
```diff
@@ -574,10 +574,10 @@
     }
 
     // init
-    let cssdom = Module.canvas.parentElement.querySelector(`#css_${object.item.key}`);
+    let cssdom = Module.canvas.parentElement?.querySelector(`#css_${object.item.key}`);
```
Comment (damian08kk) [inline src/assets/scripts/ProjectManager/Scene/HTMLElement.js:577]: Are you sure that changing assets is okay? Can you explain me, why you are changing it?

File: src/utils/restructureData.util.ts [hunk 2]
```diff
@@ -65,13 +71,28 @@ export const restructureData = (
         node,
         entity
       );
+      const newHtmlHud = createNewHtmlHud(primitiveKey, node, entity)
 
       if (node.children && node.children?.length > 0) {
-        const [tree, configurations, config] = deepTreeIteration(node.children);
+        const [tree, configurations, htmlHud, config, _hasHtmlHud] = deepTreeIteration(node.children);
```
Comment (damian08kk) [inline src/utils/restructureData.util.ts:77]: Why `_hasHtmlHud` instead of just `hasHtmlHud`?

**Review by damian08kk — CHANGES_REQUESTED** [review]

**Comment by damian08kk** [comment]
I gave you request changes because I want to know answer to question about assets :)

File: src/utils/restructureData.util.ts (hunk 2 above)
Comment (mwojcik-sdh) [reply src/utils/restructureData.util.ts:77]: `hasHtmlHud` is already taken above

File: src/assets/scripts/ProjectManager/Scene/HTMLElement.js (hunk 1 above)
Comment (mwojcik-sdh) [reply src/assets/scripts/ProjectManager/Scene/HTMLElement.js:577]: So the problem is that when you are trying to import project that has htmlHud inside object-group, whole app breaks. 
The same behaviour can be seen in old editor - basically you are stuck on loading screen forever and there are errors in console about not being able to get `querySelector` of null.

These changes make sure that import won't be stuck on loading but will proceed.
Of course, you won't have htmlHud in object-group anyway, but that's something that will need an answer from Walid.

So anyway, there is a high chance that assets would need some changes to be able to import projects with htmlHud in object-group.

**Review by damian08kk — APPROVED** [review]

**Pull request merged**